File: donne/src/main/java/freelance/paiement/donne/models/Partner.java
package freelance.paiement.donne.models;

import Enum.EtatPartner;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.Email;
import java.util.List;

@Entity
@Getter
@Setter
public class Partner {
    @Id
    @GeneratedValue
    private Long id;
    private String nom;
    private String identifiant;
    private String adresse;
    @Email
    private String email;
    @Enumerated
    private EtatPartner etat;
    @JsonIgnore
    @OneToMany
    private List<Paiement> paiements;
    @OneToOne(cascade = CascadeType.ALL)
    private Photo logo;
    @OneToOne(cascade = CascadeType.ALL)
    private Photo contrat;
    @OneToOne(cascade = CascadeType.PERSIST)
    private Compte compte;
}

File: backoffice/src/app/services/data/Partner.ts
import {EtatPartner} from '../Enum/EtatPartner';
import {Paiement} from './Paiement';
import {Photo} from './Photo';
import {Compte} from './Compte';

export class Partner {
  id: number;
  nom: string;
  identifiant: string;
  email: string;
  adresse: string;
  logo: Photo;
  etat: EtatPartner;
  paiements: Array<Paiement>;
  compte: Compte;
  contrat: Photo;
}
